File: src/main/java/com/matag/game/status/GameStatusRepository.java
package com.matag.game.status;

import java.util.HashMap;
import java.util.Map;

import org.springframework.stereotype.Component;

import com.matag.game.security.SecurityHelper;

@Component
public class GameStatusRepository {
  private final SecurityHelper securityHelper;

  public GameStatusRepository(SecurityHelper securityHelper) {
    this.securityHelper = securityHelper;
  }

  private Map<String, GameStatus> games = new HashMap<>();

  public boolean contains(String gameId) {
    return games.containsKey(gameId);
  }

  public void save(String gameId, GameStatus gameStatus) {
    games.put(gameId, gameStatus);
  }

  public GameStatus getUnsecure(String gameId) {
    return games.get(gameId);
  }

  public GameStatus get(String gameId, String sessionId) {
    var gameStatus = games.get(gameId);
    securityHelper.isPlayerAllowedToExecuteAction(gameStatus, sessionId);
    return gameStatus;
  }

  public void clear() {
    games.clear();
  }
}

File: src/main/kotlin/com/matag/game/status/GameStatusRepository.kt
package com.matag.game.status

import com.matag.game.security.SecurityHelper
import org.springframework.stereotype.Component

@Component
class GameStatusRepository(private val securityHelper: SecurityHelper) {
    private val games: MutableMap<String, GameStatus> = HashMap()

    fun contains(gameId: String): Boolean {
        return games.containsKey(gameId)
    }

    fun save(gameId: String, gameStatus: GameStatus) {
        games[gameId] = gameStatus
    }

    fun getUnsecure(gameId: String): GameStatus? {
        return games[gameId]
    }

    fun get(gameId: String, sessionId: String): GameStatus {
        val gameStatus = games[gameId]!!
        securityHelper.isPlayerAllowedToExecuteAction(gameStatus, sessionId)
        return gameStatus
    }

    fun clear() {
        games.clear()
    }
}
